### DISCUSSION TITLE
Do not display 0

### DISCUSSION THREAD

**Discussion by beagle-elgaeb** [jaredpalmer/formik#3434] [Q&A] [answered]
How to make the input have the initial value 0, type "number", but nothing would be displayed in the input itself (so that the placeholder would be visible)?

**Comment by johnrom** [comment, marked as answer]
Number inputs should be typed `"" | number`. That is, empty should be a possible value (whether or not it validates is another question).

A user might actually input "0", so that should generally be distinguishable from an empty input, plus an input populated with 0 makes it harder to enter the first number (you cannot click the input and type "1", you must double click it or press delete and then type 1).

If you're accepting "" and want it cast to 0, you can do something like:

```
onSubmit={values => {
  submit(values.numInput || 0);
}}
```

**Reply by beagle-elgaeb** [reply]
Thank you for your help!
